From bc2a531b708835c954bc2bda6e3cb7cd7ae0ff01 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:44:25 -0400
Subject: [x86] pci: Drop pcibios_scan_root() check for bus already scanned

Message-id: <20140806034425.1438.31518.stgit@gir.stowe>
Patchwork-id: 87412
O-Subject: [RHEL7.1 PATCH 03/69] x86/PCI: Drop pcibios_scan_root() check for bus already scanned
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: f19e84824ac62d662a1d2b2b367f23b98bbd6d1d

commit f19e84824ac62d662a1d2b2b367f23b98bbd6d1d
Author: Bjorn Helgaas <bhelgaas@google.com>
Date:   Tue Jan 28 16:28:21 2014 -0700

    x86/PCI: Drop pcibios_scan_root() check for bus already scanned

    The PCI core checks to see whether we've already scanned a bus, so we don't
    need to do it in pcibios_scan_root().  Here's where it happens in the core:

      pcibios_scan_root
        pci_scan_bus_on_node
          pci_scan_root_bus
            pci_create_root_bus
          b2 = pci_find_bus(pci_domain_nr(b), bus)
          if (b2)
            goto err_out;    # already scanned this bus

    This removes the check from pcibios_scan_root().

    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    Acked-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/pci/common.c b/arch/x86/pci/common.c
index 981c2db..c47bb22 100644
--- a/arch/x86/pci/common.c
+++ b/arch/x86/pci/common.c
@@ -458,15 +458,6 @@ void __init dmi_check_pciprobe(void)
 
 struct pci_bus *pcibios_scan_root(int busnum)
 {
- struct pci_bus *bus = NULL;
-
- while ((bus = pci_find_next_bus(bus)) != NULL) {
-  if (bus->number == busnum) {
-   /* Already scanned */
-   return bus;
-  }
- }
-
  return pci_scan_bus_on_node(busnum, &pci_root_ops,
      get_mp_bus_to_node(busnum));
 }
-- 
1.7.1